test/HealthChecks.System.Tests/Functional/FolderHealthCheckTests.cs

```diff
@@ -0,0 +1,96 @@
+using System.Net;
+using FluentAssertions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace HealthChecks.System.Tests.Functional
+{
+    public class folder_healthcheck_should
+    {
+        [Fact]
+        public async Task be_healthy_if_folder_is_available()
+        {
+            var webHostBuilder = new WebHostBuilder()
+               .ConfigureServices(services =>
+               {
+                   services.AddHealthChecks()
+                       .AddFolder(setup =>
+                        setup.Folder = Directory.GetCurrentDirectory(), tags: new string[] { "folder" });
+               })
+               .Configure(app =>
+               {
+                   app.UseHealthChecks("/health", new HealthCheckOptions
+                   {
+                       Predicate = r => r.Tags.Contains("folder")
+                   });
+               });
+
+            using var server = new TestServer(webHostBuilder);
+
+            var response = await server.CreateRequest($"/health")
+                .GetAsync();
+
+            response.StatusCode
+                .Should().Be(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task be_healthy_if_no_folder_provided()
```

Comment:
Are you sure about such behavior?

Reply:
Same comment from earlier (return healthy is no folder provided), but I'm open.